Tests/ImportDumpCommandTest.php

```diff
@@ -0,0 +1,205 @@
+
+        $serverUrl = $this->protector->getServerUrl();
+
+        Http::fake([
+            $serverUrl => Http::response(__FUNCTION__, 200, ['Chunk-Size' => 100]),
+        ]);
+
+        $this->artisan('protector:import --remote')
+            ->expectsConfirmation($this->shouldImportDump);
+
+        $this->assertFileExists($this->disk->path('protector/remote_dump.sql'));
+    }
+
+    /**
+     * @test
+     */
+    public function canGetRemoteDumpWithFlushOptionEnabled()
+    {
+        $this->provideDynamicDumps(['dump.sql', 'secondDump.sql' => 'dump.sql', 'thirdDump.sql' => 'dump.sql']);
+
+        Config::set('protector.remoteEndpoint.htaccessLogin', '1234:1234');
+        Config::set('protector.routeMiddleware', []);
+
+        Http::fake([
+            $this->serverUrl => Http::response(__FUNCTION__, 200, ['Chunk-Size' => 100]),
+        ]);
+
+        $this->artisan('protector:import --remote --flush')
+            ->expectsConfirmation($this->shouldImportDump);
+
+        $this->assertEquals([sprintf('%s%sremote_dump.sql', Config::get('protector.baseDirectory'), DIRECTORY_SEPARATOR)], $this->protector->getDumpFiles());
+    }
+
+    /**
+     * @test
+     */
+    public function failGetRemoteOnDumpWithNoResponse()
+    {
+        Http::fake([
+            __FUNCTION__ => Http::response(),
+        ]);
+
+        $this->artisan('protector:import --remote')
+            ->expectsOutput("Error retrieving dump from remote server: Could not fetch database from remote server: The scheme '' is not supported.")
+            ->assertFailed();
+    }
+
+    /**
+     * @test
+     */
+    public function failOptionFileOnNonExistingDump()
+    {
+        $this->provideDynamicDumps(['dump.sql']);
+
+        $fileName = 'thisDumpDoesNotExist.sql';
+
+        $this->artisan(sprintf('protector:import --file=%s --ignore-connection-filter --force', $fileName))
+            ->expectsOutput(sprintf('The file "%s" was not found.', $fileName));
+    }
+
+    /**
+     * @test
+     */
+    public function failOptionDumpOnNonExistingDump()
+    {
+        $this->expectException(FileNotFoundException::class);
+
+        $this->artisan('protector:import --dump=thisFileDoesNotExist');
+    }
+
+    /**
+     * @test
+     */
+    public function canImportDumpOnOptionLatest()
+    {
+        $this->provideDynamicDumps(['dump.sql']);
+
+        $this->artisan('protector:import --latest')->expectsConfirmation($this->shouldImportDump);
+        $this->assertEquals(sprintf('%s%sdump.sql', $this->protector->getBaseDirectory(), DIRECTORY_SEPARATOR), $this->protector->getLatestDumpName());
+    }
+
+    /**
+     * @test
+     */
+    public function failChooseImportDumpOnNoFilesInBaseDirectory()
+    {
+        $this->provideDynamicDumps([]);
+
+        $this->expectException(EmptyBaseDirectoryException::class);
+
+        $this->artisan('protector:import')
+            ->expectsChoice($this->shouldDownloadDump, 2, ['Download remote Dump']);
```

Comment:
Second choice is missing I think.

Reply:
The import dump choice isn't needed here, the exception is thrown before the user even gets the chance to choose which dump to import.